### PR TITLE
create centralized deployer lib and use it for all deployers

### LINKED ISSUE #12: Deployer Controller Library
Currently every deployer implements basic functionality on their own and every improvement has to be done in every deployers code.
Also every new deployer has to comply to the landscaper contract of interacting with the deploy items which is a big burden for developers who want to contribute a deployer.
Especially with the current state of the documentation.

Therefore, it would be beneficial to have, similar to the gardener extensions library, a deployer library that implements the landscaper-deployitem-api and offer convenient helper functions.

SharedFunctionality:
- when to reconcile (reconcile annotation or spec change)
- when to delete (check for deletion timestamp)
- add finalizers during first reconconcile and remove them when the deletion succeeds
- decode provider configuration
- encode and decode provider status
- enhanced update behavior(only update deploy item if items are updated)
- [optional] multi cluster support -> host cluster/landscaper cluster)

### PR DESCRIPTION
**Pull request by schrodit**
**How to categorize this PR?**

/area deployers
/kind cleanup
/priority 2

**What this PR does / why we need it**:

currently 90% of a deployers code is copy pasted boilerplate.
With that its also hard to keep up with updated contracts or bugfixes.

This PR creates a new deployer controller lib that exposes a simpler interface for deployer where the default contract to the landscaper is automatically handled by the lib.

**Which issue(s) this PR fixes**:
Fixes #12 

**Special notes for your reviewer**:

**Release note**:

```feature developer
A new deployer lib has been created that automatically implements the interaction with Landscaper.
Deployer do now only have to implement a simple interface with their business logic.
All current deployers are already updated to use the new deployer lib.
```

### CODE REVIEW

File: cmd/manifest-deployer-controller/app/app.go [hunk 1]
```diff
@@ -7,29 +7,24 @@ package app
...
 func NewManifestDeployerControllerCommand(ctx context.Context) *cobra.Command {
 	options := NewOptions()
 
 	cmd := &cobra.Command{
-		Use:   "manifest-deployer",
-		Short: "Manifest Deployer is a controller that applies kubernetes manifests based on DeployItems of type Manifest",
-
-		Run: func(cmd *cobra.Command, args []string) {
+		Use:          "manifest-deployer",
+		Short:        "Manifest Deployer is a controller that applies kubernetes manifests based on DeployItems of type Manifest",
```
[the remark this answers is not in the document]
Comment (schrodit) [reply cmd/manifest-deployer-controller/app/app.go:21]: you mean the `applies kubernetes manifests of type <manifest>` part?

[the remark this answers is not in the document]
Comment (achimweigel) [reply cmd/manifest-deployer-controller/app/app.go:21]: yes

[the remark this answers is not in the document]
Comment (Diaphteiros) [reply cmd/manifest-deployer-controller/app/app.go:21]: I agree, it is probably more helpful to put the precise type here.

**Review by Diaphteiros — APPROVED** [review]
/lgtm

**Pull request merged**